### PR TITLE
feat(events): add extracted fields based on req/res types

### PR DESCRIPTION
**Pull request by lsampras**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [x] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Add a trait to allow adding top level fields & types for each request
- these fields are now collected from either the response or request in order
- This are exposed as top level fields allowing to query over multiple types of api events with a single tag

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

This will allow us to query multiple events with a single filter allowing some type of tag based searches
for e.g you can filter all the payment related api calls with a single payment id filter

## How did you test it?

- Make a payment create api call & check logs

image

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ivor-juspay — DISMISSED** [review]

**Review by harsh-sharma-juspay — APPROVED** [review]

**Review by harsh-sharma-juspay — DISMISSED** [review]

File: Makefile [hunk 1]
```diff
@@ -65,7 +65,7 @@ fmt :
 #	make clippy
 
 clippy :
-	cargo clippy --all-features --all-targets -- -D warnings
+	cargo clippy --all-features --all-targets -- -D warnings -A clippy::large_futures
```
Comment (NishantJoshi00) [inline Makefile:68]: Shouldn't we handle large futures?

File: crates/common_utils/src/lib.rs [hunk 2]
```diff
@@ -6,6 +6,8 @@ pub mod consts;
 pub mod crypto;
 pub mod custom_serde;
 pub mod errors;
+#[allow(missing_docs)] // Todo: add docs
```
Comment (NishantJoshi00) [inline crates/common_utils/src/lib.rs:9]: Please create a issue or a PR to add documentation for this

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]
Other than that, looks good!

**Review by SanchithHegde — APPROVED** [review]

**Review by likhinbopanna — APPROVED** [review]

**Comment by lsampras** [comment]
force-merging as conditions met

**Pull request merged**